packages/hardhat-core/src/internal/hardhat-network/provider/HardhatBlockchain.ts

```diff
@@ -62,9 +95,15 @@ export class HardhatBlockchain implements HardhatBlockchainInterface {
     if (actual === undefined) {
       throw new Error("Invalid block");
     }
-    const nextBlock = this._data.getBlockByNumber(
-      new BN(actual.header.number).addn(1)
-    );
+
+    const nextBlockNumber = new BN(actual.header.number).addn(1);
```

Comment:
`deleteLaterBlocks` is updated in `HardhatBlockchain` but not in `ForkBlockchain`. Is that intentional?

Reply:
No, not intentional, thank you for pointing this out.  I updated `ForkBlockchain` in https://github.com/nomiclabs/hardhat/pull/2032/commits/29992fce6ff842d6b51d71ed3013471fdade222b, but then also significantly re-worked both in https://github.com/nomiclabs/hardhat/pull/2032/commits/e27cf777fd160db2a4323217d495d07e26e874cd